### PR TITLE
Cleanup of get_reverse_indexed_cases

### PR DESCRIPTION
**Pull request by esoergel**
To be merged after https://github.com/dimagi/commcare-hq/pull/18989

### CODE REVIEW

**Comment by emord** [comment]
@esoergel fyi if you want to make this a bit easier to review before the other PR is merged, you can click "edit" next to the PR header and change the branch it's based off of

File: corehq/sql_proxy_accessors/sql_templates/get_reverse_indexed_cases.sql [hunk 1]
```diff
@@ -1,6 +0,0 @@
-DROP FUNCTION IF EXISTS get_reverse_indexed_cases(TEXT, TEXT[]);
```
Comment (emord) [inline corehq/sql_proxy_accessors/sql_templates/get_reverse_indexed_cases.sql]: This function will still exist on all environments which could be confusing. Might want another migration that just contains this line with a drop. (This processes is really annoying :/ )

Comment (esoergel) [reply corehq/sql_proxy_accessors/sql_templates/get_reverse_indexed_cases.sql]: Oh you're right - I meant to add that in.  Will do so.

**Comment by esoergel** [comment]
@emord updated

File: corehq/form_processor/README.md [hunk 2]
```diff
@@ -67,9 +67,40 @@ If the function signature is changing:
...
+
+        operations = [
+            HqRunSQL("DROP FUNCTION IF EXISTS get_reverse_indexed_cases(TEXT, TEXT[]);"),
+        ]
+    ```
+2. Delete the `.sql` files containing the old function definition.
+3. Replace all references to the old file in existing migrations with `noop_migration`
     1. Or you can just remove it completely if there are other migration operations
     happening in the migration.
+5. Repeat the above for the pl_proxy function defined in `sql_proxy_accessors`
```
Comment (emord) [inline corehq/form_processor/README.md]: 1, 2, 3, 5

**Review by emord — APPROVED** [review]

File: corehq/form_processor/README.md (hunk 2 above)
Comment (esoergel) [reply corehq/form_processor/README.md]: Luckily markdown is [smart enough to handle that!](https://github.com/dimagi/commcare-hq/blob/8f2d65d9f5e942f407104a50dbf07bce5d54b950/corehq/form_processor/README.md)  I shoulda just done 1, 1, 1, 1, or #, #, #, #.  Updated anyways.

**Comment by emord** [comment]
@esoergel I think this can be merged now? May want to rebase it on master and let tests run to make sure it doesn't conflict with any new migrations though

**Comment by snopoke** [comment]
@esoergel bump

**Comment by esoergel** [comment]
@emord @snopoke sorry about that, should be g2g now.  Just waiting on the build.

**Pull request merged**